### PR TITLE
Thirdy Party Authentication with passport

### PR DESCRIPTION
**Pull request by harshithpabbati**
- GitHub
- Google

Frontend part: https://github.com/codeuino/social-platform-donut-frontend/pull/342

### CODE REVIEW

File: app/models/User.js [hunk 1]
```diff
@@ -28,7 +28,6 @@ const UserSchema = new mongoose.Schema(
     password: {
       type: String,
       trim: true,
-      required: true,
       minlength: 6,
       validate (password) {
         if (password.toLowerCase().includes('password')) {
```
Comment (vaibhavdaren) [inline app/models/User.js]: .

File: app/routes/index.js [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
+      res.redirect('http://localhost:3000/dashboard');
+  }
+);
+
+router.get('/auth/github',
+  passport.authenticate('github', { scope: [ 'user:email' ] }));
+
+router.get('/auth/github/callback', 
+  passport.authenticate('github', { 
+    failureRedirect: 'http://localhost:3000'
```
Comment (vaibhavdaren) [inline app/routes/index.js]: .

File: app/routes/index.js [hunk 3]
```diff
@@ -1,9 +1,38 @@
...
+   }),
+  (req, res) => {
+    res.redirect('http://localhost:3000/dashboard');
```
Comment (vaibhavdaren) [inline app/routes/index.js]: .

File: config/authTokens.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+    'githubAuth' : {
+        'clientID'      : process.env.GITHUBCLIENTID,
+        'clientSecret'  : process.env.GITHUBSECRET,
+        'callbackURL'   : 'http://localhost:4000/auth/github/callback'
+    },
+    'googleAuth' : {
+        'clientID'      : process.env.GOOGLECLIENTID,
+        'clientSecret'  : process.env.GOOGLECLIENTSECRET,
+        'callbackURL'   : 'http://localhost:4000/auth/google/callback'
```
Comment (vaibhavdaren) [inline config/authTokens.js]: .

File: config/authTokens.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    }
+};
```
Comment (vaibhavdaren) [inline config/authTokens.js]: .

**Comment by devesh-verma** [comment]
@harshithpabbati dont hardcode failure redirect to localhost:3000, because when this goes to production there will be no localhost and this will fail.
I feel it should be '/' which is where we have login.

**Comment by devesh-verma** [comment]
@harshithpabbati also, please check why travis is failing.

**Comment by devesh-verma** [comment]
@harshithpabbati also, the login which you have added to `app/routes/index.js` should be move to `auth`

**Comment by devesh-verma** [comment]
@harshithpabbati hey can you please resolve the conflict.

**Comment by devesh-verma** [comment]
@harshithpabbati hey are you still willing to work on this?

**Pull request closed without merging**